IFComp/lib/Plack/Middleware/IFComp.pm

```diff
@@ -90,9 +90,15 @@ sub call {
                     my $session_ref = thaw( decode_base64( $session_row->session_data ) );
                     if ( $session_ref->{ __user } ) {
                         my $user_id = $session_ref->{ __user }->{ id };
+                        my $user =
+                            $self->schema->resultset( 'User' )->find( $user_id )
+                        ;
                         if ( $user_id == $entry->author->id ) {
                             $current_user_can_see_this_game = 1;
                         }
+                        elsif ( grep { $_->name eq 'curator' } $user->roles->all ) {
```

Comment:
I'd like to see $user checked for definedness here, particularly given the long-lived (1 year?) sessions.